src/prefect/tasks/monte_carlo/client.py

```diff
@@ -0,0 +1,256 @@
+import json
+import pendulum
+import requests
+from typing import Any, Dict, List
+
+from prefect.utilities.logging import get_logger
+
+
+class MonteCarloClient:
+    def __init__(
+        self,
+        api_key_id: str,
+        api_token: str,
+    ) -> None:
+        self.api_key_id = api_key_id
+        self.api_token = api_token
+        self.logger = get_logger()
+        self._api_url = "https://api.getmontecarlo.com/graphql"
+
+    def graphql(self, query: str, variables: dict = None) -> Dict[str, Any]:
```

Comment:
I recommend renaming this function to include a verb so that it's more descriptive. I'd also prefix it with a `_` since it seems like this is a private function. Something like `_perform_graphql_operation`